internal/pkg/template/workload.go

```diff
@@ -756,21 +794,18 @@ type WorkloadOpts struct {
 	DependsOn                map[string]string
 	Publish                  *PublishOpts
 	ServiceDiscoveryEndpoint string
-	HTTPVersion              *string
 	ALBEnabled               bool
-	HostedZoneAliases        AliasesForHostedZone
 	CredentialsParameter     string
 	PermissionsBoundary      string
-	HTTPRedirect             bool
 
 	// Additional options for service templates.
 	WorkloadType            string
 	HealthCheck             *ContainerHealthCheck
 	HTTPTargetContainer     HTTPTargetContainer
 	HTTPHealthCheck         HTTPHealthCheckOpts
 	DeregistrationDelay     *int64
-	AllowedSourceIps        []string
 	NLB                     *NetworkLoadBalancer
+	ALB                     *ALBListener
```

Comment:
Same for this `ALB`. `NLB` is `NetworkLoadBalancer` whereas `ALB` is  `ALBListener`

For ALB there are ALB level config and listener level config. I think the struct should be like this
```golang
type ApplicationLoadBalancer struct {
   LBSetting
   Listeners []listenerconfig
}

type listenerconfig struct {
   ...
}
```

Reply:
I agree to your point of renaming `ALB` to `ALBListener` as it holds listener configurations. And for the structure that you are suggesting above:
1. We have only one listener and multiple listener rules so I think there is no need to have `Listeners []listenerconfig`.  
2. The reason why we got rid of the separate `LBSetting` and `Listeners` because of this [comment](https://github.com/aws/copilot-cli/pull/4501#discussion_r1113481634) that we discussed earlier. 

So I think just renaming `ALB` to `ALBListeners` would do as per our expectations. Let me know what do you feel about it.